oci/casext/walk.go

```diff
@@ -176,19 +176,19 @@ func (e Engine) Paths(ctx context.Context, root ispec.Descriptor) ([]DescriptorP
 // Walk(). The returned slice will *not* contain any duplicate digest.Digest
 // entries. Note that without descriptors, a digest is not particularly
 // meaninful (OCI blobs are not self-descriptive).
-func (e Engine) Reachable(ctx context.Context, root ispec.Descriptor) ([]digest.Digest, error) {
-	seen := map[digest.Digest]struct{}{}
+func (e Engine) Reachable(ctx context.Context, root ispec.Descriptor) ([]ispec.Descriptor, error) {
+	seen := map[digest.Digest]ispec.Descriptor{}
```

Comment:
And this is incorrect because you can have multiple descriptors (with different values) pointing to the same digest.

Reply:
Hmm...the issue is then is that the policy may "ok" to GC for one descriptor and later "not ok" but it is now too late.